CMakeLists.txt

```diff
@@ -30,27 +36,38 @@ CPMAddPackage(
   VERSION 1.4
 )
 
+find_package(fmt)
+if(NOT TARGET fmt::fmt)
+  # FIXME this add a target without namespace! CK
+  CPMAddPackage(
+    NAME fmt
+    GIT_TAG 7.1.3
+    GITHUB_REPOSITORY fmtlib/fmt
+  )
+endif()
+
 # ---- Add source files ----
 
 # Note: globbing sources is considered bad practice as CMake's generators may not detect new files
 # automatically. Keep that in mind when changing files, or explicitly mention them here.
-file(GLOB_RECURSE headers CONFIGURE_DEPENDS "${CMAKE_CURRENT_SOURCE_DIR}/include/*.h")
-file(GLOB_RECURSE sources CONFIGURE_DEPENDS "${CMAKE_CURRENT_SOURCE_DIR}/source/*.cpp")
+file(GLOB_RECURSE headers CONFIGURE_DEPENDS "include/*.h")
+file(GLOB_RECURSE sources CONFIGURE_DEPENDS "source/*.cpp")
 
 # ---- Create library ----
 
 # Note: for header-only libraries change all PUBLIC flags to INTERFACE and create an interface
-# target: add_library(Greeter INTERFACE) set_target_properties(Greeter PROPERTIES
-# INTERFACE_COMPILE_FEATURES cxx_std_17)
-
-add_library(Greeter ${headers} ${sources})
+# target:
+add_library(Greeter)
+add_library(${PROJECT_NAME}::Greeter ALIAS Greeter)
+target_compile_features(Greeter PUBLIC cxx_std_17)
 
-set_target_properties(Greeter PROPERTIES CXX_STANDARD 17)
+target_sources(Greeter PRIVATE ${headers} ${sources})
```

Comment:
What is the advantage of `target_sources` vs adding them at the library definition?

Reply:
the library target may created in on `CMakeLists.txt`

and the source be added later in a sub project added with `add_subdirectory()`